### PR TITLE
Make all Cassiopeia variables !default

### PR DESCRIPTION
**Pull request by RickR2H**
Pull Request for Issue # .

### Summary of Changes
In this PR I I added the "!default" setting to all the Cassiopeia variables. There where some variables who had already this setting in place, but now all variables are set as default. Setting the"!default" makes is easier for developers to create custom styling based on the Cassiopeia template for child themes for instance. 

**Use case:**
After a child theme is created, create in the media/templates/site/CHILD_THEME a /scss folder
in this folder create a template.scss with the following content:
```
$secondary: red;
$success: red;
$info: red;
$warning: red;
$danger: red;

@import "../../cassiopeia/scss/template.scss";
// Your extra custom styling below
```
Next copy the template-rtl.scss from the media/templates/site/cassiopeia/scss folder to the child theme /scss folder.
Compile the template.scss and template-rtl.scss to the /css folder of the child theme
After compiling the CSS is build and the variable overrides are respected.
Set the child theme under the template settings in the backend and see the result.

### Testing Instructions
Checkout this PR and compile the styling with npm ci.
Everything should work a normal.

### CODE REVIEW

**Comment by ricardo1709** [comment]
I have tested this item :white_check_mark: successfully on ab8a55eed24d25ae89cf1a230efab670e4d247b9<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/37360">issues.joomla.org/tracker/joomla-cms/37360</a>.</sub>

File: build/media_source/templates/site/cassiopeia/scss/tools/variables/_variables.scss [hunk 1]
```diff
@@ -133,34 +133,34 @@ $colors: (
...
+$link-hover-color:                    var(--cassiopeia-color-hover) !default;
 
 // Breadcrumbs
-$breadcrumb-active-color:             $gray-700;
+$breadcrumb-active-color:             $gray-700 !default;
 
 // Grid columns
-$grid-gutter-width:                   1em !default;
-$cassiopeia-grid-gutter:              $grid-gutter-width;
+$grid-gutter-width:                   1em !default !default;
```
Comment (Quy) [inline build/media_source/templates/site/cassiopeia/scss/tools/variables/_variables.scss]:
```suggestion
$grid-gutter-width:                   1em !default;
```

Comment (RickR2H) [reply build/media_source/templates/site/cassiopeia/scss/tools/variables/_variables.scss]: Thanks @Quy

**Comment by MSnoeren** [comment]
I have tested this item :white_check_mark: successfully on 59da1f7e89db505fb7682dc4d69d4a04dc10bafa<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/37360">issues.joomla.org/tracker/joomla-cms/37360</a>.</sub>

**Comment by ricardo1709** [comment]
I have tested this item :white_check_mark: successfully on 59da1f7e89db505fb7682dc4d69d4a04dc10bafa<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/37360">issues.joomla.org/tracker/joomla-cms/37360</a>.</sub>

**Comment by RickR2H** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/37360">issues.joomla.org/tracker/joomla-cms/37360</a>.</sub>

**Comment by laoneo** [comment]
Thanks!

**Pull request merged**